torch/distributed/_functional_collectives.py

```diff
@@ -192,6 +192,9 @@ def all_gather_tensor(
     res = _maybe_wrap_tensor(tensor)
     # TODO this should be done inside AsyncCollectiveTensor to delay the wait() call
     if gather_dim != 0:
+        # torch.cat access the data so we already need to wait here, first do wait
+        # and then chunk + cat avoid us going through ACT dispatching logic again
+        res = res.wait()  # type: ignore[attr-defined]
```

Comment:
iirc, this is only a performance optimization. No logic is changed.

Reply:
Yep.